src/admin/runners/fixInstantDataTypeRunner.js

```diff
@@ -0,0 +1,232 @@
+     */
+    async processRecordAsync (doc) {
+        try {
+            const operations = [];
+            /**
+             * @type {Resource}
+             */
+            let resource = FhirResourceCreator.create(doc);
+            /**
+             * @type {Resource}
+             */
+            const currentResource = resource.clone();
+
+            const dateColumnHandler = new DateColumnHandler();
+            dateColumnHandler.setFlag(true);
+            resource = await dateColumnHandler.preSaveAsync({ resource: resource });
+
+            // for speed, first check if the incoming resource is exactly the same
+            const updatedResourceJsonInternal = resource.toJSONInternal();
+            const currentResourceJsonInternal = currentResource.toJSONInternal();
+
+            if (deepEqual(updatedResourceJsonInternal, currentResourceJsonInternal) === true) {
+                return operations;
+            }
+            operations.push({
+                replaceOne: {
+                    filter: {
+                        _id: doc._id
+                    },
+                    replacement: updatedResourceJsonInternal
+                }
+            });
+
+            return operations;
+        } catch (e) {
+            throw new RethrownError(
+                {
+                    message: `Error processing record ${e.message}`,
+                    error: e,
+                    args: {
+                        resource: doc
+                    },
+                    source: 'FixInstantDataTypeRunner.processRecordAsync'
+                }
+            );
+        }
+    }
+
+    /**
+     * main process function
+     * @returns {Promise<void>}
+     */
+    async processAsync () {
+        try {
+            if (this.collections.length > 0 && this.collections[0] === 'all') {
+                /**
+                 * @type {string[]}
+                 */
+                this.collections = await this.getAllCollectionNamesAsync();
+                this.collections = this.collections.sort();
+                if (this.startFromCollection) {
+                    this.collections = this.collections.filter(
+                        (c) => c >= this.startFromCollection
+                    );
+                }
+            }
+            const mongoConfig = await this.mongoDatabaseManager.getClientConfigAsync();
+
+            for (const collectionName of this.collections) {
+                const startFromIdContainer = this.createStartFromIdContainer();
+
+                const resource = collectionName.substring(0, collectionName.indexOf('_'));
+                if (!(resourcesWithDateTimeFields(resource))) {
+                    this.adminLogger.logInfo(`${collectionName} has no dateTime fields. Skipping`);
+                    continue;
+                }
+                this.adminLogger.logInfo(`Processing ${collectionName}`);
+                const uuids = [];
+                while (uuids.length > 0) {
```

Comment:
This while loop won't execute as uuids is an empty list

Reply:
restored missing code